golang/pkg/builder/container/utils.go

```diff
@@ -138,15 +138,31 @@ func pullImage(ctx context.Context, logger io.StringWriter, fullyQualifiedImageN
 }
 
 func deleteContainer(ctx context.Context, containerName string) error {
-	if err := stopContainer(ctx, containerName); err != nil {
-		return err
+	deletableContainer, err := getContainer(ctx, containerName)
+	if err != nil {
+		return fmt.Errorf("builder could not get container (%s) to remove: %s", containerName, err.Error())
 	}
 
-	if err := removeContainer(ctx, containerName); err != nil {
-		return err
+	switch deletableContainer.State {
+	case "":
+		// when there's no container we just skip it
+		return nil
+	case "running":
+		if err = stopContainer(ctx, containerName); err != nil {
+			return fmt.Errorf("builder could not stop container (%s): %s", containerName, err.Error())
+		}
+		fallthrough
+	case "exited":
```

Comment:
What about the other states? For example, if a container is `dead` or `paused`? I think we have to implement all states.

Reply:
Oh, yeah, and restarting.... good catch!

Reply:
Or created 🪄

Reply:
fixed :sparkles: